kiwi_stackbuild_plugin/tasks/system_stash.py

```diff
@@ -0,0 +1,115 @@
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+"""
+usage: kiwi-ng system stash -h | --help
+       kiwi-ng system stash --root=<directory>
+           [--tag=<name>]
+       kiwi-ng system stash help
+
+commands:
+    stash
+        create a container from the given root directory
+    stash help
+        show manual page for stash command
+
+options:
+    --root=<directory>
+        the path to the root directory, usually the result of
+        a former system prepare or build call
+    --tag=<name>
+        the tag name for the container.
+        By default set to: latest
+"""
+import os
+import logging
+
+from kiwi.help import Help
+from kiwi.tasks.base import CliTask
+from kiwi.privileges import Privileges
+from kiwi.xml_description import XMLDescription
+from kiwi.xml_state import XMLState
+from kiwi.oci_tools import OCI
+from kiwi.path import Path
+
+from kiwi_stackbuild_plugin.defaults import StackBuildDefaults
+
+log = logging.getLogger('kiwi')
+
+
+class SystemStashTask(CliTask):
+    def process(self) -> None:
+        if self.command_args.get('help') is True:
+            self.manual = Help()
+            return self.manual.show('kiwi::system::stash')
+
+        Privileges.check_for_root_permissions()
+
+        log.info('Reading Image description')
+        description = XMLDescription(
+            os.path.join(self.command_args['--root'], 'image', 'config.xml')
+        )
+        xml_state = XMLState(
+            xml_data=description.load()
+        )
+        contact_info = xml_state.get_description_section()
+        stash_target_dir = SystemStashTask._create_stash_target_dir(
+            self.command_args['--root'],
+            xml_state.xml_data.get_name()
+        )
+
+        stash_container_file_name = os.path.join(
+            stash_target_dir, 'stash.tar'
+        )
+        stash_container_tag = self.command_args['--tag'] or 'latest'
+        container_config = StackBuildDefaults.get_container_config(
+            os.path.basename(stash_container_file_name),
+            stash_container_tag,
+            contact_info.author
+        )
+        log.info('Initializing stash container')
+        oci = OCI.new()
+        if os.path.exists(stash_container_file_name):
+            log.info('--> Adding new layer on existing stash')
+            oci.import_container_image(
+                f'oci-archive:{stash_container_file_name}:{stash_container_tag}'
+            )
+        else:
+            log.info('--> Creating initial layer')
+            oci.init_container()
+
+        oci.unpack()
+        oci.sync_rootfs(
+            self.command_args['--root'],
+            StackBuildDefaults.get_stash_exclude_list()
+        )
+        oci.repack(container_config)
+        oci.post_process()
+        log.info('Exporting stash container')
+        oci.export_container_image(
+            stash_container_file_name, 'oci-archive', stash_container_tag
```

Comment:
I'd love to be capable to push and fetch images from remote registries and I believe by adding some more flexibility here this should be possible. However I am fine with this code to start playing with this plugin.

Imagine 
```
kiwi-ng system stack --root=<dir> [--tag=<tag>] [--repository=<registry_repo>]
```

`registry_repo` could be something like `quay.io/mypublicrepo/myappliance` and if your host (`skopeo`) is configured with the credentials to quay.io the image should be pushed as `quay.io/mypublicrepo/myappliance:tag`.

Reply:
good idea. I'll add an issue and I think this could be added later as a followup change

Reply:
I added https://github.com/OSInside/kiwi-stackbuild-plugin/issues/2